src/OpenTelemetry/Metrics/InstrumentIdentity.cs

```diff
@@ -0,0 +1,74 @@
+// <copyright file="InstrumentIdentity.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+
+namespace OpenTelemetry.Metrics
+{
+    internal readonly struct InstrumentIdentity : IEquatable<InstrumentIdentity>
+    {
+        public InstrumentIdentity(string meterName, string instrumentName, string unit, string description, Type instrumentType)
+        {
+            this.MeterName = meterName;
+            this.InstrumentName = instrumentName;
+            this.Unit = unit;
+            this.Description = description;
+            this.InstrumentType = instrumentType;
+        }
+
+        public readonly string MeterName { get; }
```

Comment:
This is not quite right. The entire meter - that is, name, version, and schema url (when we eventually support it) should be a component of the identity. Fixing this...

Reply:
Fixed https://github.com/open-telemetry/opentelemetry-dotnet/pull/2916/commits/03a245e2815adbd5f4efa09827a7b2c98b04f074

Reply:
> This is not quite right. The entire meter - that is, name, version, and schema url (when we eventually support it) should be a component of the identity. Fixing this...

Wouldn't it be better to just have `Meter` as the only `public` property in that case? We would only need to update the `Equals()` check like below:
```
                ...
                && this.Meter.Name == other.Meter.Name
                && this.Meter.Version == other.Meter.Version
                && this.Meter.SchemaUrl == other.Meter.SchemaUrl 
                ...
```
This way if schema url gets added to `Meter` later on, we don't have to add a new `public` property dedicated just for that.

Reply:
Since this is an `internal` `struct`, it wouldn't matter as much if we have to add more properties later on, but I think we could just use `Meter` here as it provides the best encapsulation for everything `Meter` related.

Reply:
Do you mean keep a handle on the `Meter` itself? Same concern here https://github.com/open-telemetry/opentelemetry-dotnet/pull/2916#discussion_r819248034 would apply. `Meter` might be disposed so safer to not keep a handle on it.

Reply:
Maybe it'd be less of a concern to keep a handle on the meter since `InstrumentIdentity` is internal?

Reply:
Yeah, I was suggesting to have a handle to `Meter` itself as all of this is happening in the `InstrumentPublished` callback so the `Meter` would not be disposed in this path. But I overlooked the fact that in case of dictionary lookup collisions, the `Equals` check might fail as it might still have entries to instruments whose `Meter`s are disposed. This would only work if we can ensure that the dictionary would never have any instrument whose `Meter` is disposed.

Reply:
Okay to park this into an issue and come back to this ? (so that we can merge and do a release.)